source/custom/nixnet_service.custom.cpp

```diff
@@ -1,2 +1,168 @@
+#include <nixnet/nixnet_service.h>
+#include <server/converters.h>
+
+#include <atomic>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
 namespace nixnet_grpc {
+
+// Returns true if it's safe to use outputs of a method with the given status.
+inline bool status_ok(int32 status)
+{
+  return status >= 0;
+}
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+::grpc::Status NiXnetService::GetProperty(::grpc::ServerContext* context, const GetPropertyRequest* request, GetPropertyResponse* response)
+{
+  if (context->IsCancelled()) {
+    return ::grpc::Status::CANCELLED;
+  }
+  try {
+    auto session_ref_grpc_session = request->session_ref();
+    nxSessionRef_t session_ref = session_repository_->access_session(session_ref_grpc_session.id(), session_ref_grpc_session.name());
+    u32 property_id;
+    switch (request->property_id_enum_case()) {
+      case nixnet_grpc::GetPropertySizeRequest::PropertyIdEnumCase::kPropertyId: {
+        property_id = static_cast<u32>(request->property_id());
+        break;
+      }
+      case nixnet_grpc::GetPropertySizeRequest::PropertyIdEnumCase::kPropertyIdRaw: {
+        property_id = static_cast<u32>(request->property_id_raw());
+        break;
+      }
+      case nixnet_grpc::GetPropertySizeRequest::PropertyIdEnumCase::PROPERTY_ID_ENUM_NOT_SET: {
+        return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The value for property_id was not specified or out of range");
+        break;
+      }
+    }
+
+    u32 property_size{};
+    auto status = library_->GetPropertySize(session_ref, property_id, &property_size);
+    if (!status_ok(status)) {
+      response->set_status(status);
+      return ::grpc::Status::OK;
+    }
+
+    switch (property_type_map_[property_id]) {
```

Comment:
You'll need to make sure and test this thoroughly. There's not codesharing that helps validate this logic is correct.